### PR TITLE
feat: add fancy theme-toggle button

### PR DESCRIPTION
**Pull request by abrahamcalsin**

### CODE REVIEW

File: src/components/dark-theme-button/components/moon-fill-icon/moon-fill-icon.tsx [hunk 1]
```diff
@@ -1,34 +1,35 @@
...
   const { className } = props
 
   return (
     <motion.svg
-      className={className}
-      width="16px"
-      height="16px"
-      fill="currentColor"
-      viewBox="0 0 20 20"
+      className={`w-full ${className}`}
```
Comment (yonycalsin) [inline src/components/dark-theme-button/components/moon-fill-icon/moon-fill-icon.tsx:20]: I recommend using a utility for constructing className strings conditionally.

for do that, you can use [clsx](https://www.npmjs.com/package/clsx)

```suggestion
      className={clsx(`w-full`, className)}
```

File: src/components/dark-theme-button/components/sun-fill-icon/sun-fill-icon.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  initial: { opacity: 0, scale: 0.8, originX: 'center', originY: 'center', rotate: -80 },
+  animate: { opacity: 1, scale: 1, rotate: 0, transition: MOTION_TRANSITION },
+}
+
+export function SunFillIcon(props: SunFillIconProps) {
+  const { className } = props
+
+  return (
+    <motion.svg
+      className={`w-full dark:text-secondary ${className}`}
```
Comment (yonycalsin) [inline src/components/dark-theme-button/components/sun-fill-icon/sun-fill-icon.tsx:26]: also here.

```suggestion
      className={clsx(`w-full dark:text-secondary`, className)}
```

File: src/components/dark-theme-button/dark-theme-button.tsx [hunk 3]
```diff
@@ -1,29 +1,32 @@
...
+import { useTheme } from 'next-themes'
 
 import { MoonFillIcon } from './components/moon-fill-icon'
-import { SunStrokeIcon } from './components/sun-stroke-icon'
+import { SunFillIcon } from './components/sun-fill-icon'
 
 export function DarkThemeButton() {
-  const { onToggle } = useDarkTheme()
+  const [mounted, setMounted] = React.useState(false)
+  const { systemTheme, theme, setTheme } = useTheme()
```
Comment (yonycalsin) [inline src/components/dark-theme-button/dark-theme-button.tsx:10]:
```suggestion
  const [mounted, setMounted] = React.useState(false)
  
  const { systemTheme, theme, setTheme } = useTheme()
```

**Review by yonycalsin — CHANGES_REQUESTED** [review]

File: src/components/dark-theme-button/dark-theme-button.tsx [hunk 4]
```diff
@@ -1,29 +1,32 @@
...
-    position: 'absolute',
-    transition: 'transform 300ms ease-in-out',
-  }
+  const currentTheme = theme === 'system' ? systemTheme : theme
+
+  React.useEffect(() => {
+    setMounted(true)
+  }, [])
+
+  if (!mounted) return null
```
Comment (yonycalsin) [inline src/components/dark-theme-button/dark-theme-button.tsx:18]: good hack 🌮 🚀 

https://www.npmjs.com/package/next-themes#avoid-hydration-mismatch

File: src/components/dark-theme-button/components/moon-fill-icon/moon-fill-icon.tsx (hunk 1 above)
Comment (abrahamcalsin) [reply src/components/dark-theme-button/components/moon-fill-icon/moon-fill-icon.tsx:20]: great, already changed, thanks 🎉

File: src/components/dark-theme-button/components/sun-fill-icon/sun-fill-icon.tsx (hunk 2 above)
Comment (abrahamcalsin) [reply src/components/dark-theme-button/components/sun-fill-icon/sun-fill-icon.tsx:26]: cool, thanks

File: src/components/dark-theme-button/dark-theme-button.tsx (hunk 3 above)
Comment (abrahamcalsin) [reply src/components/dark-theme-button/dark-theme-button.tsx:10]: thanks 👍

File: src/components/dark-theme-button/dark-theme-button.tsx (hunk 4 above)
Comment (abrahamcalsin) [reply src/components/dark-theme-button/dark-theme-button.tsx:18]: yeah of course 😁🎉

**Review by yonycalsin — APPROVED** [review]
Good Job 🎉

**Pull request merged**